Switchbot blind tilt support (#26127)
home-assistant/home-assistant.github.io: source/_integrations/switchbot.markdown

A SwitchBot lock can be set up in Home Assistant in two different ways. You can enter the key id and encryption key yourself, or Home Assistant can import them from your SwitchBot account.

## SwitchBot Blind Tilt

> The blind tilt is exposed as a cover entity with control of the tilt position only. Open is represented as 50% with closed down as 0% and closed up as 100%. Because Home Assistant believes 100% is open, the default cards will disable the open button when the tilt is at 100%, but the service call will still work and open the blind to 50%.

Reviewer: Its unfortunate we don't have a better way to model this, but that would likely require an architecture change beyond the the scope of this.

> The blind tilt is exposed as a cover entity with control of the tilt position only. Open is represented as 50% with closed down as 0% and closed up as 100%. Because Home Assistant believes 100% is open, the default cards will disable the open button when the tilt is at 100%, but the service call will still work and open the blind to 50%.
> 

Reviewer 2: ````suggestion
The blind tilt is exposed as a cover entity with control of the tilt position only:
| Tilt Position | Blind State |
| ------------- | ----------- |
| 100%          | Closed Up   |
| 50%           | Fully Open  |
| 0%            | Closed Down |

The close button will close the blinds to the closest closed position (either 0% or 100%), and defaults to closing down if the blinds are fully open. Because Home Assistant believes 100% is open, the default cards will disable the open button when the tilt is at 100%, but the service call will still work and open the blind to 50%.

### Simple cover template entity

Some integrations may expose your SwitchBot Blind Tilt to other services which expect that 100% is open and 0% is fully closed. Using a [Cover Template](/integrations/cover.template), a proxy entity can be created which will be open at 100% and closed at 0%. This template entity is limited to closing in one direction.

```
# Example configuration.yaml entry
cover:
  - platform: template
    covers:
      example_blinds_simple:
        device_class: blind
        friendly_name: Example Blinds (Simple Down)
        open_cover:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: 50
          target:
            entity_id: cover.example_blinds
        close_cover:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: 0
          target:
            entity_id: cover.example_blinds
        position_template: >
          {{ int(states.cover.example_blinds.attributes.current_tilt_position)*2 }}
        set_cover_position:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: "{{position/2}}"
          target:
            entity_id: cover.example_blinds
```

````
Take whatever you want from this (or nothing). The template example is pretty long for a main doc page and might be better suited for a forum post.

Reviewer 2: Take whatever you want from this (or nothing). The template example is pretty long for a main doc page and might be better suited for a forum post.

Reviewer: Given we currently don't have a good way to model the tilt blinds (this is a pretty unique device) I think the template makes sense here. If other integrations end up support these type of devices we can do an architecture change to better support these in the underlying model and remove the template.

A SwitchBot lock can be set up in Home Assistant in two different ways. You can enter the key id and encryption key yourself, or Home Assistant can import them from your SwitchBot account.

## SwitchBot Blind Tilt

The blind tilt is exposed as a cover entity with control of the tilt position only:
| Tilt Position | Blind State |
| ------------- | ----------- |
| 100%          | Closed Up   |
| 50%           | Fully Open  |
| 0%            | Closed Down |

The close button will close the blinds to the closest closed position (either 0% or 100%), and defaults to closing down if the blinds are fully open. Because Home Assistant believes 100% is open, the default cards will disable the open button when the tilt is at 100%, but the service call will still work and open the blind to 50%.

### Simple cover template entity

Some integrations may expose your SwitchBot Blind Tilt to other services which expect that 100% is open and 0% is fully closed. Using a [Cover Template](/integrations/cover.template), a proxy entity can be created which will be open at 100% and closed at 0%. This template entity is limited to closing in one direction.

> 

Reviewer: ```suggestion

{% raw %}

```

> ```

Reviewer: ```suggestion
```yaml
```

# Example configuration.yaml entry
cover:
  - platform: template
    covers:
      example_blinds_simple:
        device_class: blind
        friendly_name: Example Blinds (Simple Down)
        open_cover:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: 50
          target:
            entity_id: cover.example_blinds
        close_cover:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: 0
          target:
            entity_id: cover.example_blinds
        position_template: >
          {{ int(states.cover.example_blinds.attributes.current_tilt_position)*2 }}
        set_cover_position:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: "{{position/2}}"
          target:
            entity_id: cover.example_blinds
```

> 

Reviewer: ```suggestion
{% endraw %}

```

Some integrations may expose your SwitchBot Blind Tilt to other services which expect that 100% is open and 0% is fully closed. Using a [Cover Template](/integrations/cover.template), a proxy entity can be created which will be open at 100% and closed at 0%. This template entity is limited to closing in one direction.

{% raw %}

```yaml
# Example configuration.yaml entry
cover:
  - platform: template
    covers:
      example_blinds_simple:
        device_class: blind
        friendly_name: Example Blinds (Simple Down)
        open_cover:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: 50
          target:
            entity_id: cover.example_blinds
        close_cover:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: 0
          target:
            entity_id: cover.example_blinds
        position_template: >
          {{ int(states.cover.example_blinds.attributes.current_tilt_position)*2 }}
        set_cover_position:
          service: cover.set_cover_tilt_position
          data:
            tilt_position: "{{position/2}}"
          target:
            entity_id: cover.example_blinds
```

> {% endraw %}

Reviewer: ```suggestion

{% endraw %}
```

A SwitchBot lock can be set up in Home Assistant in two different ways. You can enter the key id and encryption key yourself, or Home Assistant can import them from your SwitchBot account.

## SwitchBot Blind Tilt

The blind tilt is exposed as a cover entity with control of the tilt position only:

> | Tilt Position | Blind State |

Reviewer: ```suggestion

| Tilt Position | Blind State |
```
